### PR TITLE
Revamp authorisation to support returning information and batching

### PR DESCRIPTION
**Pull request by a3957273**

### CODE REVIEW

**Comment by a3957273** [comment]
We haven't updated or created any tests for this, so it won't pass inspection, but consideration over the approach would be useful.

**Review by JR40159 — DISMISSED** [review]

File: backend/src/connectors/v2/authorisation/base.ts [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+          }
+        }
+
+        return { success: true, id: access.name }
+      }),
+    )
+  }
+}
+
+// type Response = { success: true; info: undefined } | { success: false; info: string }
```
Comment (ARADDCC002) [inline backend/src/connectors/v2/authorisation/base.ts]: Is there a reason this is commented out? If so can we add a "todo" to this part of the code, ideally with a Jira ticket if possible.

File: backend/src/scripts/migrateV2.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      roles: ['msro', 'owner'],
+    })
+  }
+
+  for (const reviewer of model.latestVersion.metadata.contacts.reviewer) {
+    const entity = `${reviewer.kind}:${identityConversion(reviewer.id)}`
+
+    const search = collaborators.find((entry) => entry.entity === entity)
+    if (search) {
+      search.roles.push('mtr')
```
Comment (ARADDCC002) [inline backend/src/scripts/migrateV2.ts:73]: Are these role values `mtr` etc typed anywhere?

File: backend/src/services/v2/model.ts [hunk 3]
```diff
@@ -116,13 +119,14 @@ export async function searchModels(
     }
   }
 
-  const results = ModelModel
+  const results = await ModelModel
     // Find only matching documents
```
Comment (ARADDCC002) [inline backend/src/services/v2/model.ts:123]: I don't know if these comments really add much, it seems quite self explanatory to me

**Review by ARADDCC002 — APPROVED** [review]

**Pull request merged**